backend/server.ts

```diff
@@ -0,0 +1,65 @@
+import express from 'express';
+import http from 'http';
+import { Server } from 'socket.io';
+
+const app: express.Express = express();
+const server: http.Server = http.createServer(app);
+const io: Server = new Server(server);
+
+const players: { [key: number]: Player } = {};
+
+app.use(express.static(__dirname + "/public"));
+
+app.get("/", function (req: express.Request<any>, res: express.Response<any>) {
+  res.sendFile(__dirname + "/index.html");
+});
+
+io.on("connection", function (socket: any) {
+  console.log("a user connected");
+
+  // create a new player and add it to our players object
+  players[socket.id] = {
+    rotation: 0,
+    x: Math.floor(Math.random() * 700) + 50,
+    y: Math.floor(Math.random() * 500) + 50,
+    playerId: socket.id,
+  };
+
+  // send the players object to the new player
+  socket.emit("currentPlayers", players);
+
+  // update all other players of the new player
+  socket.broadcast.emit("newPlayer", players[socket.id]);
+
+  // when a player disconnects, remove them from our players object
+  socket.on("disconnected", function () {
```

Comment:
```suggestion
  socket.on("disconnect", function () {
```

client が切断した際自動的に```"disconnect"```イベントが発火される為、それを server でキャッチする必要があると思います（現在プレイヤーが切断しても他の client に反映されていなかったため）